### PR TITLE
Add celo to networks

### PR DESCRIPTION
**Pull request by GigaHierz**
## Description

We would like to add Celo as a network to the Scaffold ETH template, before the ETH Global hackathon, but also because we have received a few requests from our community! 
We love this project and our community does at well! thank you for your work! 

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #{#979}_  (currently opened as a discussion item)

_Note: If your changes are small and straightforward, you may skip the creation of an issue beforehand and remove this section. However, for medium-to-large changes, it is recommended to have an open issue for discussion and approval prior to submitting a pull request._

Your ENS/address: gigahierz.eth

### CODE REVIEW

**Comment by carletex** [comment]
Hey @GigaHierz , thanks for this.

We usually don't add new chains (since we just provide some basic ones and you can add whatever network you like), but after talking with @technophile-04 we thought that Celo could be a nice addition.

So let's test and merge.

File: packages/hardhat/hardhat.config.ts [hunk 1]
```diff
@@ -121,6 +121,14 @@ const config: HardhatUserConfig = {
       url: "https://sepolia.publicgoods.network",
       accounts: [deployerPrivateKey],
     },
+    celo: {
+      url: "https://forno.celo.org",
+      accounts: [deployerPrivateKey],
+    },
+    celoAlfajores: {
+      url: "https://forno.alfajores.celo-testnet.org/",
```
Comment (technophile-04) [inline packages/hardhat/hardhat.config.ts:129]: I think this RPC url doesn't work? 

tried doing `yarn deploy --network celoAlfajores`



Should it be https://alfajores-forno.celo-testnet.org ?

File: packages/nextjs/utils/scaffold-eth/networks.ts [hunk 2]
```diff
@@ -31,6 +31,8 @@ export const RPC_CHAIN_NAMES: Record<number, string> = {
   [chains.base.id]: "base-mainnet",
   [chains.baseGoerli.id]: "base-goerli",
   [chains.baseSepolia.id]: "base-sepolia",
+  [chains.celo.id]: "celo-mainnet",
+  [chains.celoAlfajores.id]: "celo-alfajores",
 };
```
Comment (technophile-04) [inline packages/nextjs/utils/scaffold-eth/networks.ts:36]: So this mapping is only used for creating alchemy url's, I think Alchemy doesn't support celo right? I couldn't find the celo rpc url on my alchemy dashboard

Comment (GigaHierz) [reply packages/nextjs/utils/scaffold-eth/networks.ts:36]: no. Unfortunately, it does not support Celo yet. Is that needed to be added?

**Review by technophile-04 — APPROVED** [review]
This looks great! Thanks @GigaHierz!

**Pull request merged**